File: apps/ios/Fini/Design/Tokens.swift
import SwiftUI

extension Color {
    // Base
    static let finiCream       = Color(hex: "#F7F5E8")
    static let finiCreamDark   = Color(hex: "#EFEADA")
    static let finiInk         = Color(hex: "#1F1A15")
    static let finiInkSoft     = Color(hex: "#4A4237")
    static let finiInkMute     = Color(hex: "#8A8070")
    static let finiLine        = Color(hex: "#1F1A15").opacity(0.08)
    static let finiLineSoft    = Color(hex: "#1F1A15").opacity(0.05)
    // Chat / Rose
    static let finiRose        = Color(hex: "#FDDDE8")
    static let finiRoseDeep    = Color(hex: "#F9C6D7")
    static let finiPink        = Color(hex: "#C85A8A")
    static let finiPinkSoft    = Color(hex: "#E89AB3")
    // Insights / Mint
    static let finiMint        = Color(hex: "#D6EEE0")
    static let finiMintDeep    = Color(hex: "#BEE3CB")
    static let finiGreen       = Color(hex: "#5B8E6F")
    static let finiGreenSoft   = Color(hex: "#A7C9B4")
    // Data / Lavender
    static let finiLavender    = Color(hex: "#E8ECFF")
    static let finiLavenderDeep = Color(hex: "#D4DBFA")
    static let finiBlue        = Color(hex: "#5A6FB8")
    static let finiBlueSoft    = Color(hex: "#A5B0D8")
    // Finance
    static let finiIncome      = Color(hex: "#5B8E6F")
    static let finiIncomeTint  = Color(hex: "#DDEEDF")
    static let finiExpense     = Color(hex: "#D47070")
    static let finiExpenseTint = Color(hex: "#FADEDC")
    static let finiGold        = Color(hex: "#C9A24D")
    static let finiGoldTint    = Color(hex: "#F3E7C7")
}

extension Color {
    init(hex: String) {
        let hex = hex.trimmingCharacters(in: CharacterSet.alphanumerics.inverted)
        var int: UInt64 = 0
        Scanner(string: hex).scanHexInt64(&int)
        let a, r, g, b: UInt64
        switch hex.count {
        case 3: (a, r, g, b) = (255, (int >> 8) * 17, (int >> 4 & 0xF) * 17, (int & 0xF) * 17)
        case 6: (a, r, g, b) = (255, int >> 16, int >> 8 & 0xFF, int & 0xFF)
        case 8: (a, r, g, b) = (int >> 24, int >> 16 & 0xFF, int >> 8 & 0xFF, int & 0xFF)
        default: (a, r, g, b) = (255, 0, 0, 0)
        }
        self.init(.sRGB, red: Double(r)/255, green: Double(g)/255, blue: Double(b)/255, opacity: Double(a)/255)
    }
}

extension Font {
    static func rubik(_ size: CGFloat, weight: Font.Weight = .regular) -> Font {
        return .system(size: size, weight: weight, design: .default)
    }
}

struct FiniTokens {
    static let radiusSm: CGFloat = 10
    static let radiusMd: CGFloat = 16
    static let radiusLg: CGFloat = 22
    static let radiusXl: CGFloat = 28
}

File: packages/design-system/tokens.ts
export const FINI = {
  // base
  cream: '#F7F5E8',
  creamDark: '#EFEADA',
  ink: '#1F1A15',
  inkSoft: '#4A4237',
  inkMute: '#8A8070',
  line: 'rgba(31,26,21,0.08)',
  lineSoft: 'rgba(31,26,21,0.05)',
  // rose (chat)
  rose: '#FDDDE8',
  roseDeep: '#F9C6D7',
  pink: '#C85A8A',
  pinkSoft: '#E89AB3',
  // mint (insights)
  mint: '#D6EEE0',
  mintDeep: '#BEE3CB',
  green: '#5B8E6F',
  greenSoft: '#A7C9B4',
  // lavender (data)
  lavender: '#E8ECFF',
  lavenderDeep: '#D4DBFA',
  blue: '#5A6FB8',
  blueSoft: '#A5B0D8',
  // finance semantics
  income: '#5B8E6F',
  incomeTint: '#DDEEDF',
  expense: '#D47070',
  expenseTint: '#FADEDC',
  gold: '#C9A24D',
  goldTint: '#F3E7C7',
  // typography
  font: '"Rubik", -apple-system, system-ui, sans-serif',
  // radii
  radiusSm: 10,
  radiusMd: 16,
  radiusLg: 22,
  radiusXl: 28,
  // shadows
  shadowCard: '0 1px 2px rgba(31,26,21,0.04)',
  shadowLifted: '0 4px 16px rgba(31,26,21,0.06)',
  shadowSheet: '0 -8px 30px rgba(31,26,21,0.15)',
  shadowPink: '0 8px 24px rgba(200,90,138,0.3)',
  shadowGold: '0 2px 6px rgba(201,162,77,0.3)',
};

export type TabId = 'chat' | 'insights' | 'data';
export type ScreenId = 'chat' | 'brief' | 'addTx' | 'insights' | 'savings' | 'data' | 'transactions' | 'couples' | 'paywall' | 'txDetail' | 'obWelcome' | 'obGoal' | 'obIncome' | 'obPlan';
export type PlanId = 'free' | 'pro' | 'proPlus';
